limbus/components/kornia.contrib.py

```diff
@@ -0,0 +1,154 @@
+"""Some predefined components for kornia.
+
+All the components are "saved" as classes in limbus.components, being the original module name used
+to create a clear name for the component.
+E.g.:
+limbus.components.base.ImageReader -> limbus.components.limbus.base.ImageReader
+torch.unbind                       -> limbus.components.torch.unbind
+kornia.color.rgb_to_hls            -> limbus.components.kornia.color.rgb_to_hls
+
+The components automatically created are always saved in the limbus.components module.
+"""
+from typing import Any, List, Optional
+import logging
+
+import cv2
+import numpy as np
+import torch
+import kornia
+
+from limbus.core import Component, ComponentState, Params, WidgetState
+from limbus import widgets
+
+
+log = logging.getLogger(__name__)
+
+
+class ShowFaceLandmarks(widgets.WidgetComponent):
+    """Component that draw face landmarks on an image.
+
+    Args:
+        name (str): component name.
+
+    Inputs:
+        image (torch.Tensor): a batch of images (NxCxHxW).
+        landmarks (torch.Tensor): a batch of landmarks (Nx15).
+
+    Viz params:
+        title (str): title of the window. Default: "".
+
+    """
+    # the viz state by default is disabled but can be enabled by the user with the widget_state property.
+    WIDGET_STATE: WidgetState = WidgetState.DISABLED
+
+    def __init__(self, name: str):
+        super().__init__(name)
+
+    @staticmethod
+    def register_inputs(inputs: Params) -> None:  # noqa: D102
+        inputs.declare("image", torch.Tensor)
+        inputs.declare("landmarks", torch.Tensor)
+
+    @staticmethod
+    def register_properties(properties: Params) -> None:  # noqa: D102
+        # this line is like super() but for static methods.
+        widgets.WidgetComponent.register_properties(properties)  # adds the title param
+        properties.declare("nrow", Optional[int], value=None)
+        properties.declare("draw_keypoints", bool, False)
+        properties.declare("threshold", float, 0.8)
+
+    def _draw_keypoint(self, img: np.ndarray,
+                       det: kornia.contrib.FaceDetectorResult,
+                       kpt_type: kornia.contrib.FaceKeypoint) -> np.ndarray:
+        kpt = det.get_keypoint(kpt_type).int().tolist()
+        return cv2.circle(img, kpt, 2, (255, 0, 0), 2)
+
+    def _draw_landmarks(self, image: torch.Tensor, landmarks: torch.Tensor) -> torch.Tensor:
+        dets: List[kornia.contrib.FaceDetectorResult] = [kornia.contrib.FaceDetectorResult(o) for o in landmarks]
+        frame_vis: np.ndarray = kornia.tensor_to_image(image).copy()
+        frame_vis = (frame_vis * 255).astype(np.uint8)
+        for b in dets:
+            if b.score < self._properties.get_param("threshold"):
```

Comment:
```suggestion
            if b.score < self._properties.get_property("threshold"):
```
this is more intuitive since it's declared as property

Reply:
or just `self.get_property`

Reply:
the __call__ option I think it is the best one. get_param cannot be changed easily since it is a method inside a the param class.

Reply:
not doable in an easy way. Anyway I do not see a big issue with this.